gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +164,227 @@ def _get_prior_measurements_in_world_frame(
+        tracks_to_cameras_ratio: float = TRACKS_TO_CAMERAS_RATIO,
+    ) -> List[SfmTrack2d]:
+        """Removes bad tracks and returns the longest ones from the rest based on tracks_to_cameras_ratio.
+
+        Bad tracks are those that have fewer than 3 measurements from valid_cameras.
+        Sorts the remaining tracks in descending order by number of measurements, and returns as many tracks
+        as tracks_to_cameras_ratio * number of valid cameras.
+
+        Args:
+            tracks: List of all input tracks.
+            valid_cameras: Set of valid camera indices (these have direction measurements and valid rotations).
+            intrinsics: List of camera intrinsics.
+            tracks_to_cameras_ratio: Ratio of tracks to cameras to use for averaging.
+
+        Returns:
+            List of tracks to use for averaging.
+        """
+        max_tracks = int(len(valid_cameras) * tracks_to_cameras_ratio)
+        filtered_tracks = []
+        valid_cameras_with_intrinsics = set([c for c in valid_cameras if intrinsics[c] is not None])
+        for track in tracks:
+            valid_cameras_track = track.select_for_cameras(camera_idxs=valid_cameras_with_intrinsics)
+            if valid_cameras_track.number_measurements() < 3:
+                continue
+            filtered_tracks.append(valid_cameras_track)
+        return sorted(filtered_tracks, key=lambda track: track.number_measurements(), reverse=True)[:max_tracks]
+
+    def _get_landmark_directions(
+        self,
+        tracks_2d: List[SfmTrack2d],
+        intrinsics: List[Optional[gtsfm_types.CALIBRATION_TYPE]],
+        wRi_list: List[Optional[Rot3]],
+    ) -> RelativeDirectionsDict:
+        """Computes the camera to landmark directions for each track, in world frame.
+
+        Args:
+            tracks_2d: 2d tracks in each image, assuming that all measurements in all tracks are for valid cameras.
+            intrinsics: camera intrinsics for each camera.
+            wRi_list: camera rotations in world frame.
+
+        Returns:
+            Dictionary of unit directions from camera to track in world frame indexed by (track_id, camera_id).
+        """
+        landmark_directions = {}
+        for track_id, track in enumerate(tracks_2d):
+            for j in range(track.number_measurements()):
+                measurement = track.measurement(j)
+                cam_idx = measurement.i
+
+                if intrinsics[cam_idx] is None or wRi_list[cam_idx] is None:
+                    raise ValueError("Camera intrinsics or rotation cannot be None for input track measurements")
+
+                measurement_xy = intrinsics[cam_idx].calibrate(measurement.uv)
+                measurement_homog = Point3(measurement_xy[0], measurement_xy[1], 1.0)
+                w_cam_U_track = Unit3(wRi_list[cam_idx].rotate(Unit3(measurement_homog).point3()))
+
+                landmark_directions[(track_id, cam_idx)] = w_cam_U_track  # w_i2Ui1 here
```

Comment:
is the commented out line intentional? Can we add a few more words if so to make it clearer?